scripts/commands/jail.lua

```diff
@@ -26,31 +27,60 @@ commandObj.onTrigger = function(player, target, cellId, reason)
         { -620, -400, -620, 0 },  { -180, -400, -620, 0 }, { 260, -400, -620, 0 }, { 700, -400, -620, 0 },
     }
 
-    -- Validate the target..
-    local targ = GetPlayerByName(target)
-    if targ == nil then
-        player:printToPlayer(string.format('Invalid player \'%s\' given.', target))
-        return
+    -- defaults
+    local targetIsOnline = false
+    local targetID = 0
+
+    -- find the target object (this only works on targets on the same map process (cluster) as GM)
+    local targetObj = GetPlayerByName(target)
+
+    -- did not find the target (so either invalid target name, offline target, or online target in a different map process (cluster) than GM)
+    if targetObj == nil then
+        -- check if the target name is in the database
+        targetID = GetPlayerIDByName(target)
+        -- if name in the database then will have valid ID > 0
+        if targetID > 0 and targetID < 0xFFFFFFFF then
+            -- next check if the target has a valid session (and thus online in a different map process (cluster))
+            if PlayerHasValidSession(targetID) then
+                -- target online in different map process (cluster) than GM so inform the GM and exit
+                player:printToPlayer(string.format("Player '%s' is online but in a different zone. Go to that zone to use !jail", target))
```

Comment:
I think you meant to say cluster instead of zone.

Reply:
The wording here would confuse or missinform of the actual command usage in the other direction.

GMs SHOULD know how the server is setup

Reply:
> The wording here would confuse or missinform of the actual command usage in the other directiion.
> 
> GMs SHOULD know how the server is setup
> 
> I guess you could change it to "zone group" (since cluster would also be technically incorrect from what i've been told)

Yeah, I will change to zone group or something, thanks for feedback! 

Yeah ideally GMs would know but for example there are over 50 "clusters" on Horizon so I think GMs would need to consult a table and the clustering is changing over time for performance reasons.